library/posture.c

```diff
@@ -905,29 +905,46 @@ static bool check_running(uv_loop_t *loop, const char *path) {
 #if _WIN32
     HANDLE sh = CreateToolhelp32Snapshot(TH32CS_SNAPPROCESS, 0);
     if (sh == INVALID_HANDLE_VALUE) {
-        ZITI_LOG(ERROR, "failed to get process list: %d", GetLastError());
+        ZITI_LOG(ERROR, "failed to get process list: %lu", GetLastError());
         return result;
     }
-    PROCESSENTRY32W entry = {
-            .dwSize = sizeof(PROCESSENTRY32W)
-    };
+
+    // Set the size of the structure before using it.
+    PROCESSENTRY32 pe32;
+    pe32.dwSize = sizeof( PROCESSENTRY32 );
+    // Retrieve information about the first process, and exit if unsuccessful
+    if( !Process32First( sh, &pe32 ) )
+    {
+        CloseHandle( sh );          // clean the snapshot object
+        return( FALSE );
+    }
 
     char fullPath[1024];
     DWORD fullPathSize;
 
-    for (BOOL ret = Process32FirstW(sh, &entry); ret; ret = Process32NextW(sh, &entry)) {
-        HANDLE ph = OpenProcess(PROCESS_QUERY_INFORMATION, FALSE, entry.th32ProcessID);
+    // Now walk the snapshot of processes, and display information about each process in turn
+    ZITI_LOG(VERBOSE, "checking to see if process is running: %s", path);
+    do
+    {
+        ZITI_LOG(VERBOSE, "process is running: %s", pe32.szExeFile);
+
+        HANDLE ph = OpenProcess(PROCESS_QUERY_LIMITED_INFORMATION, FALSE, pe32.th32ProcessID); //OpenProcess(PROCESS_QUERY_INFORMATION, FALSE, pe32.th32ProcessID);
```

Comment:
probably safe to remove the comment on this line?

Reply:
fixed. thx